src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -116,16 +120,60 @@ public static Email parseEmail(String email) throws ParseException {
     }
 
     /**
-     * Parses a String module, and trims the String.
+     * Parses a String moduleCode, and trims the String.
      *
-     * @param module String to be trimmed
+     * @param moduleCode String to be trimmed
      * @return Trimmed String
      * @throws ParseException null
      */
-    public static String parseModule(String module) throws ParseException {
-        requireNonNull(module);
-        String trimmedModule = module.trim();
-        return trimmedModule;
+    public static ModuleCode parseModuleCode(String moduleCode) throws ParseException {
+        requireNonNull(moduleCode);
+        String trimmedModuleCode = moduleCode.trim();
+        return new ModuleCode(trimmedModuleCode);
+    }
+
+    /**
+     * Parses a String semesterNo, and trims the String.
+     *
+     * @param acadYear String to be trimmed
+     * @return Trimmed String
+     * @throws ParseException null
+     */
+    public static AcadYear parseAcadYear(String acadYear) throws ParseException {
+        requireNonNull(acadYear);
+        String trimmedAcadYear = acadYear.trim();
+        return new AcadYear(trimmedAcadYear);
+    }
+
+    /**
+     * Parses a String semesterNo, and trims the String.
+     *
+     * @param semesterNo String to be trimmed
+     * @return Trimmed String
+     * @throws ParseException null
+     */
+    public static SemesterNo parseSemesterNo(String semesterNo) throws ParseException {
+        requireNonNull(semesterNo);
+        String trimmedSemesterNo = semesterNo.trim();
+
+        SemesterNo semNo = new SemesterNo(trimmedSemesterNo);
+        return semNo;
+    }
+
+    /**
+     * Parses a String nusModsLink to an {@code NusModsShareLink}.
+     *
+     * @param nusModsLink String to be parsed
+     * @return NusModsShareLink
+     * @throws ParseException null
+     */
+    public static NusModsShareLink parseNusModsLink(String nusModsLink) throws ParseException {
+        requireNonNull(nusModsLink);
+        String trimmedNusModsLink = nusModsLink.trim();
+        if (!NusModsShareLink.isValidUrl(trimmedNusModsLink)) {
+            throw new ParseException(NusModsShareLink.MESSAGE_CONSTRAINTS);
+        }
+        return new NusModsShareLink(nusModsLink);
```

Comment:
Is there a reason as to why link validation is also done at this level? My understanding is that the `NusModsShareLink` constructor also does the same validation and can throw a similar `ParseException`.

Reply:
Hmm, in this case if don't validate here and throw a ParseException, the `checkArgument()` call in the constructor will throw an IllegalArgumentException instead. Not sure what's the best approach but it's a minor inconvenience. Since it does not affect normal operations, I'll merge first for now.